fortitude/src/rules/precision/double_precision.rs

```diff
@@ -9,20 +9,34 @@ use tree_sitter::Node;
 // TODO rule to prefer 1.23e4_sp over 1.23e4, and 1.23e4_dp over 1.23d4
 
 /// ## What it does
-/// Checks for use of 'double precision' and 'double complex' types.
+/// Checks for use of `double precision` and `double complex` types.
 ///
 /// ## Why is this bad?
-/// The 'double precision' type does not guarantee a 64-bit floating point number
-/// as one might expect. It is instead required to be twice the size of a default
-/// 'real', which may vary depending on your system and can be modified by compiler
-/// arguments. For portability, it is recommended to use `real(dp)`, with `dp` set
-/// in one of the following ways:
+/// The `double precision` type does not guarantee a 64-bit floating point number
+/// as one might expect, and instead is only required to have a higher decimal
+/// precision than the default `real`, which may vary depending on your system
+/// and can be modified by compiler arguments.
+///
+/// In modern Fortran, it is preferred to use `real` and `complex` and instead set
+/// the required precision using 'kinds'. For portability, it is recommended to use
+/// `real(dp)`, with `dp` set in one of the following ways:
 ///
 /// - `use, intrinsic :: iso_fortran_env, only: dp => real64`
 /// - `integer, parameter :: dp = selected_real_kind(15, 307)`
 ///
+/// To select a `real` with the same precision as `double precision`, the kind
+/// may be set from a floating point literal constant using `d` or `D` to denote
+/// the exponent:
+///
+/// - `integer, parameter :: dp = kind(0.0d0)
```

Comment:
This would conflict with #2?

Reply:
It would, but that seems to be the advice given in Modern Fortran 2018. I imagine that line would be a strong case for the use of a `NOLINT` comment.

I'd also be happy to remove that section altogether, as recommending `selected_real_kind`, `iso_fortran_env` and `iso_c_bindings` is confusing enough without also offering a fourth option to achieve (probably) the same thing.